### PR TITLE
Adding codeowners to quantization, sparsity, ns, etc.

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #79505

The notifications for the AO-maintained codebase.
This should not be blocking, just PR/test notifications.

### CODE REVIEW

**Review by vkuzo — APPROVED** [review]

**Review by albanD — APPROVED** [review]

File: CODEOWNERS [hunk 1]
```diff
@@ -16,6 +16,19 @@
...
+# Architecture Optimization (quantization, sparsity, etc.)
+/torch/quantization/ @jerryzh168 @vkuzo
+/test/test_quantization.py @jerryzh168
+/test/ao/ @jerryzh168 @z-a-f @hdcharles
+/test/quantization/ @jerryzh168 @z-a-f
+nn/quantized/ @jerryzh168 @vkuzo
+nn/quantizable/ @jerryzh168 @z-a-f
+nn/qat/ @jerryzh168 @vkuzo
+nn/intrinsic @jerryzh168 @vkuzo
+ao/ @vkuzo
```
Comment (jerryzh168) [inline CODEOWNERS:28]: can you add ao/quantization and add my name as well

Comment (jerryzh168) [reply CODEOWNERS:28]: and why this does not start with `/`?

Comment (albanD) [reply CODEOWNERS:28]: >and why this does not start with /?

This is the same as `**/ao/*` basically. So anything that has `/ao/` in its path will get added.

**Comment by z-a-f** [comment]
@mergebot rebase

**Review by jerryzh168 — COMMENTED** [review]
I think we want to add ownership for https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized as well, can have separate owners for:
```
https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized
https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized/cpu
https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized/cuda
https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized/cudnn
```

**Comment by z-a-f** [comment]
> I think we want to add ownership for https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized as well, can have separate owners for:
> 
> ```
> https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized
> https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized/cpu
> https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized/cuda
> https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized/cudnn
> ```

Thank you very much! Will update. Please, let me know if I missed any other places?

**Comment by jerryzh168** [comment]
I guess there is a parallel place for sparsity as well: https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/ao_sparse

**Comment by jerryzh168** [comment]
also is it possible to infer the codeowners by some heuristics?

File: CODEOWNERS [hunk 2]
```diff
@@ -21,6 +21,25 @@
 /docs/source/conf.py @albanD
 /aten/src/ATen/native/tags.yaml @anjali411
 
+# Architecture Optimization (quantization, sparsity, etc.)
+/aten/src/ATen/native/ao_sparse @z-a-f @salilsdesai @kimishpatel @digantdesai
+/aten/src/ATen/native/quantized @vkuzo @jerryzh168
+/aten/src/ATen/native/quantized/cpu @vkuzo @jerryzh168
+/aten/src/ATen/native/quantized/cuda @vkuzo @jerryzh168
+/aten/src/ATen/native/quantized/cudnn @vkuzo @jerryzh168
```
Comment (z-a-f) [inline CODEOWNERS:29]: Add David here

File: CODEOWNERS [hunk 3]
```diff
@@ -21,6 +21,25 @@
 /docs/source/conf.py @albanD
 /aten/src/ATen/native/tags.yaml @anjali411
 
+# Architecture Optimization (quantization, sparsity, etc.)
+/aten/src/ATen/native/ao_sparse @z-a-f @salilsdesai @kimishpatel @digantdesai
+/aten/src/ATen/native/quantized @vkuzo @jerryzh168
+/aten/src/ATen/native/quantized/cpu @vkuzo @jerryzh168
```
Comment (z-a-f) [inline CODEOWNERS:27]: Add FBGEMM and QNNPACK here

**Pull request closed without merging**